### PR TITLE
Flutter Guide

### PR DESCRIPTION
**Pull request by shahbaz17**
plus typo fix

### CODE REVIEW

File: src/pages/guides/flutter.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+---
+
+### Setup your Web3Auth Dashboard
+
+<SetupWeb3AuthDashboard />
+
+- Add `{YOUR_APP_PACKAGE_NAME}://auth` in the **Whitelist URL** field of the Web3Auth Dashboard.
+
+### Configure Deep Link
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:68]: ios deep linking is missing here, also feels a bit abrupt for the android part even, I got confused while reading whether I'm reading android guide

File: src/pages/guides/flutter.mdx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        // highlight-next-line
+        maven { url "https://jitpack.io" } // <-- Add this line
+    }
+}
+```
+
+Then, in your app-level `build.gradle` dependencies section, add the following:
+
+```groovy
+dependencies {
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:114]: is this right? This is referencing the android sdk

File: src/pages/guides/flutter.mdx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+For this guide here, we will be talking through the [Web3Auth Flutter SDK](/sdk/flutter).
+
+### Installation
+
+#### Add Web3Auth to Gradle
+
+In your project-level `settings.gradle` or `build.gradle` file, add JitPack repository:
+
+```groovy
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:99]: do we need jitpack in flutter as well?

File: src/pages/guides/flutter.mdx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+dependencies {
+    // ...
+    implementation 'org.torusresearch:web3auth-android-sdk:2.5.0'
+}
+```
+
+#### Permissions
+
+Open your app's `AndroidManifest.xml` file and add the following permission:
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:122]: is there any way to update permissions directly from flutter configs? Do we need to individually enable internet permissions for both android and ios?

File: src/pages/guides/flutter.mdx [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+## Using the Web3Auth SDK
+
+---
+
+To use the Web3Auth SDK, you need to add the dependency of the respective platform SDK of Web3Auth to your project. To know more about the available
+SDKs, please have a look at this [documentation page](/docs/sdk).
+
+For this guide here, we will be talking through the [Web3Auth Flutter SDK](/sdk/flutter).
+
+### Installation
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:93]: ios installation is missing

File: src/pages/guides/flutter.mdx [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+  initPlatformState();
+}
+
+// Initialization
+Future<void> initPlatformState() async {
+  HashMap themeMap = new HashMap<String, String>();
+  themeMap['primary'] = "#fff000";
+
+  Uri redirectUrl;
+  if (Platform.isAndroid) {
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:163]: again I see the code is for Android, not for iOS - have we not tested that?

File: src/pages/guides/flutter.mdx [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+```dart
+import 'dart:collection';
+import 'dart:io';
+import 'dart:async';
+
+import 'package:web3auth_flutter/web3auth_flutter.dart';
+import 'package:web3auth_flutter/enums.dart';
+import 'package:web3auth_flutter/input.dart';
+import 'package:web3auth_flutter/output.dart';
```
Comment (yashovardhan) [inline src/pages/guides/flutter.mdx:149]: why do we need to import all of these packages?

**Review by yashovardhan — CHANGES_REQUESTED** [review]
I've left a few comments. Overall I feel the iOS aspect missing is the major thing about this guide. We should add iOS as well as Android aspects when it comes to flutter.

File: src/pages/guides/flutter.mdx (hunk 2 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:114]: Yes, that's correct. It uses the android SDK for the android build.

File: src/pages/guides/flutter.mdx (hunk 3 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:99]: Yes. That's how we fetch the android related files for android build.

File: src/pages/guides/flutter.mdx (hunk 4 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:122]: These are Native calls, and I believe this is the way to do that, hence Flutter their folders and files to update and override.

File: src/pages/guides/flutter.mdx (hunk 1 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:68]: The iOS part is not tested, as the iOS SDK reference is not working for Flutter. Was hoping to use the latest updates and iOS example to use here after your iOS-related tasks.

File: src/pages/guides/flutter.mdx (hunk 5 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:93]: Didn't check the iOS part.

Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:93]: Not sure about the `pod install Web3Auth`. It doestn't work inside the flutter terminal.

File: src/pages/guides/flutter.mdx (hunk 7 above)
Comment (shahbaz17) [reply src/pages/guides/flutter.mdx:149]: It is from the examples folder of Flutter SDK repo.

**Review by yashovardhan — APPROVED** [review]
LGTM

**Pull request merged**